libs/slcsp-calculator/premium-tax-credit/household/src/lib/member/member.component.ts

```diff
@@ -46,14 +46,56 @@ export class MemberComponent {
   // This should eventually be an observable
   get validMemberDetailPage(): boolean {
     const isPrimaryMember = this.memberId === 1;
-    const validDobGroup = this.memberFormGroup.get('dob')?.valid ?? false;
+    const dob = this.memberFormGroup.get('dob');
+
+    // eslint-disable-next-line unicorn/no-null
+    this.memberFormGroup.controls['dob'].setErrors(null);
+
+    if (dob?.valid) {
+      const value = dob.value;
+
+      if (Number.parseInt(value.year, 10) > 2022) {
+        this.memberFormGroup.controls['dob'].setErrors({
+          msg: 'Date of birth cannot be after the tax year',
+        });
+        return false;
+      }
+
+      const age = this.calculateAge(
+        new Date(
+          Number.parseInt(value.year, 10),
+          Number.parseInt(value.month, 10) - 1,
+          Number.parseInt(value.day, 10)
+        )
+      );
+
+      if (age > 125) {
+        this.memberFormGroup.controls['dob'].setErrors({
+          msg: 'Age cannot exceed 125 years',
+        });
+        return false;
+      }
+    } else {
+      if (dob?.touched) {
+        this.memberFormGroup.controls['dob'].setErrors({
+          msg: 'Dob is invalid',
+        });
+      }
+
+      return false;
+    }
+
```

Comment:
Unfortunately this method doesn't ensure that correct values will be present for the month and day values.

For example, I could be perfectly fine entering a value of 14/45/2022, which isn't really a valid date, but javascript will treat it as such (i.e. the value new Date returns is Fri Mar 17 2023.  I think the date checking needs to be a little more stringent.

Also, don't we need to more careful to allow them to put in years starting with 2023?  It's possible they had a child this year and wonder if they qualify for assistance.  We should probably use the current year as the comparison.

At this point I'd also recommend that we add a spec for this control/logic, given the complexity.

Reply:
> For example, I could be perfectly fine entering a value of 14/45/2022, which isn't really a valid date, but javascript will treat it as such

No, it does not treat it as valid date. we've validations set at form group level. [here](https://github.com/ideacrew/enroll-fe/blob/trunk/libs/slcsp-calculator/household-form/src/lib/form-initialization/initial-household-form.ts#L61)

Reply:
You are correct, looks good then.